src/FieldDescription/FieldDescription.php

```diff
@@ -65,19 +62,18 @@ public function describesCollectionValuedAssociation(): bool
     protected function setAssociationMapping(array $associationMapping): void
     {
         $this->associationMapping = $associationMapping;
-
-        $this->mappingType = $this->mappingType ?: $associationMapping['type'];
-        $this->fieldName = $associationMapping['fieldName'];
+        $this->mappingType = $this->mappingType ?? $associationMapping['type'];
     }
 
     protected function setFieldMapping(array $fieldMapping): void
     {
         $this->fieldMapping = $fieldMapping;
-
-        $this->mappingType = $this->mappingType ?: $fieldMapping['type'];
-        $this->fieldName = $this->fieldName ?: $fieldMapping['fieldName'];
+        $this->mappingType = $this->mappingType ?? $fieldMapping['type'];
     }
 
+    /**
+     * @psalm-suppress DocblockTypeContradiction
```

Comment:
can you add a note a about why it cant be fixed? like the others supress

Reply:
Done with https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/pull/602#discussion_r650489002